pkg/catalog/catalog.go

```diff
@@ -110,6 +111,13 @@ type Config struct {
 	Config        *config.Config
 	KVStore       *kv.StoreMessage
 	WalkerFactory WalkerFactory
+	CacheFactory  CacheFactory
+}
+
+// CacheFactory provides an abstraction for creating cache.
+// Use in tests to override
+type CacheFactory interface {
```

Comment:
Why factory is needed? we can initialize the cache to NoCache

Reply:
Done.